eZ/Publish/Core/FieldType/Tests/Integration/BinaryBase/BinaryBaseStorage/BinaryBaseStorageTest.php

```diff
@@ -0,0 +1,159 @@
+use eZ\Publish\Core\IO\Values\BinaryFileCreateStruct;
+use eZ\Publish\Core\Persistence\Legacy\Tests\TestCase;
+use eZ\Publish\SPI\FieldType\BinaryBase\PathGenerator;
+use eZ\Publish\SPI\IO\MimeTypeDetector;
+use eZ\Publish\SPI\Persistence\Content\Field;
+use eZ\Publish\SPI\Persistence\Content\FieldValue;
+use eZ\Publish\SPI\Persistence\Content\VersionInfo;
+
+class BinaryBaseStorageTest extends TestCase
+{
+    /** @var \eZ\Publish\Core\FieldType\BinaryBase\BinaryBaseStorage\Gateway|\PHPUnit\Framework\MockObject\MockObject */
+    protected $gatewayMock;
+
+    /** @var \eZ\Publish\SPI\FieldType\BinaryBase\PathGenerator|\PHPUnit\Framework\MockObject\MockObject */
+    protected $pathGeneratorMock;
+
+    /** @var \eZ\Publish\Core\IO\IOServiceInterface|\PHPUnit\Framework\MockObject\MockObject */
+    protected $ioServiceMock;
+
+    /** @var \eZ\Publish\Core\FieldType\BinaryBase\BinaryBaseStorage|\PHPUnit\Framework\MockObject\MockObject */
+    protected $storage;
+
+    protected function setUp(): void
+    {
+        $this->gatewayMock = $this->getStorageGateway();
+        $this->pathGeneratorMock = $this->createMock(PathGenerator::class);
+        $this->ioServiceMock = $this->createMock(IOServiceInterface::class);
+        $this->storage = $this->getMockBuilder(BinaryBaseStorage::class)
+            ->setMethods(null)
+            ->setConstructorArgs(
+                [
+                    $this->gatewayMock,
+                    $this->ioServiceMock,
+                    $this->pathGeneratorMock,
+                    $this->createMock(MimeTypeDetector::class),
+                ]
+            )
+            ->getMock();
+    }
+
+    protected function getContext(): array
+    {
+        return ['context'];
+    }
+
+    public function testHasFieldData(): void
+    {
+        $this->assertTrue($this->storage->hasFieldData());
+    }
+
+    /**
+     * @dataProvider providerOfFieldData
+     */
+    public function testStoreFieldData(VersionInfo $versionInfo, Field $field): void
+    {
+        $binaryFileCreateStruct = new BinaryFileCreateStruct([
+            'id' => 'qwerty12345',
+            'size' => '372949',
+            'mimeType' => 'image/jpeg',
+        ]);
+
+        $this->ioServiceMock
+            ->expects($this->once())
+            ->method('newBinaryCreateStructFromLocalFile')
+            ->will($this->returnValue($binaryFileCreateStruct));
+
+        $this->pathGeneratorMock
+            ->expects($this->once())
+            ->method('getStoragePathForField')
+            ->with($field, $versionInfo)
+            ->willReturn('image/qwerty12345.jpg');
+
+        $this->ioServiceMock
+            ->expects($this->once())
+            ->method('createBinaryFile')
+            ->with($binaryFileCreateStruct)
+            ->willReturn(new BinaryFile());
+
+        $this->storage->storeFieldData($versionInfo, $field, $this->getContext());
+
+        $this->expectNotToPerformAssertions();
```

Comment:
`        $this->expectNotToPerformAssertions();`

This line was needed as the Solar Cloud was failing because the test had no assertions.

>which should never happen since you're mocking GW! Please investigate. Wild guess: you've using integration TestCase as base for unit tests.

I'm not mocking GW, that was just bad naming, my bad.